packages/kotti-ui/source/kotti-button/types.ts

```diff
@@ -18,17 +18,32 @@ export namespace KottiButton {
 	}
 	export const sizeSchema = z.nativeEnum(Size)
 
-	export const propsSchema = z.object({
-		icon: yocoIconSchema.nullable().default(null),
+	export const sharedSchema = z.object({
 		isBlock: z.boolean().default(false),
 		isLoading: z.boolean().default(false),
 		isMultiline: z.boolean().default(false),
 		isSubmit: z.boolean().default(false),
-		label: z.string().nullable().default(null),
 		size: sizeSchema.default(Size.MEDIUM),
 		type: typeSchema.default(Type.DEFAULT),
 	})
 
+	export const iconButtonSchema = sharedSchema.extend({
```

Comment:
@carsoli I tried to use `parseSafe` but didn't work. I think is because in the schema definition `label` and `helpText` are defined as `nullable` with `null` as default value, thats why. Even if any of them are passed they are still defined as `null` so the `parseSafe` resolves in success.

Reply:
I will remove the unused code.

Reply:
That's odd.. if you pass the value of a nullable-by-default prop to the component, it shouldn't fallback to null..

In the case of iconButtonSchema, we defined `label` as `z.null()` not `nullable().default(null)` because we actually don't want to pass it

However, i also didn't think through the "implications" of this on Vue

because I don't think Vue handles such a case, so I'm not sure how it'd have internally parsed a label in the case of iconButtonSchema, and I think this is what you meant..

I guess it's much easier to do it with JS - let's not overcomplicate validation of props if there's an available solution

Thank you for removing it and resolving..